clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -83,11 +83,120 @@ static CXXRecordDecl *getKernelObjectType(FunctionDecl *Caller) {
   return (*Caller->param_begin())->getType()->getAsCXXRecordDecl();
 }
 
+// This information is from Section 4.13 of the SYCL spec
+// https://www.khronos.org/registry/SYCL/specs/sycl-1.2.1.pdf
+// This function returns false if the math lib function is not supported
```

Comment:
Not sure if I'm misinterpreting this, but this comment may be incorrect or at least doesn't seem to align with the if statement at line 245 (https://github.com/intel/llvm/pull/613/files#diff-c88a590aca6d5f8e92dd1f8b6ec05a49R245). It looks like it will diagnose an error when it returns true at the moment, but I might be getting my wires crossed a little bit here and misreading it.

Reply:
I modified both caller and callee side comments to align now.

Reply:
LGTM, thanks for the change!